### PR TITLE
Add RusTorch to Machine learning section

### PR DESCRIPTION
**Pull request by Genius-apple**
I've added RusTorch to the Machine learning section.

RusTorch is a production-ready deep learning framework written in pure Rust. It provides a PyTorch-like API, a complete autograd engine, and neural network modules ( nn , vision , text ), making it easier for Python developers to transition to Rust for ML tasks.

- Repository : https://github.com/Genius-apple/RusTorch
- Crate : https://crates.io/crates/rus-torch
It is actively maintained and documented. Thanks for considering!

### CODE REVIEW

**Comment by palfrey** [comment]
In order to attempt to be as fair as we can with the process regarding additions of new projects we're using external popularity metrics to guide this. Our current standards (as per [our CONTRIBUTING.md](https://github.com/rust-unofficial/awesome-rust/blob/main/CONTRIBUTING.md)) are at least one of: a minimum of 50 stars on Github; 2000 downloads for projects with published crates; or equivalent values in other areas (e.g. downloads in other places, star-type popularity in other repositories).

Unfortunately the project you've submitted doesn't reach any of these thresholds. If you think that we've miscounted or there is another public metric that you think we should take into account, then please re-open this PR and detail where else we should be looking.
